tests/webdriver/color/color-common.js

```diff
@@ -0,0 +1,40 @@
+function generateScript(renderMethod) {
+  var testScript = function() {
+    var box = Crafty.e('2D, $renderMethod')
+                    .attr({ x: 0, y: 0, w: 320, h: 240 });
+
+    function initial() {
+      box.addComponent('Color');
+      signalBarrier('initial');
+    }
+
+    function opaque() {
+      box.color('rgb(0, 255, 0)');
+      signalBarrier('opaque');
+    }
+
+    function transparent() {
+      box.color('rgba(0, 0, 255, 0.5)');
+      signalBarrier('transparent');
+    }
+
+    waitBarrier('initial', initial);
+    waitBarrier('opaque', opaque);
+    waitBarrier('transparent', transparent);
+  };
+
+  return testScript.toString().replace('$renderMethod', renderMethod);
```

Comment:
Is this a standard pattern for these types of tests?  Seems like it should be possible to adjust this dynamically without doing serializing to a script with text replacement, but I might not understand the limitations of the framework.